### ISSUE TITLE
wasm-interp crashes when wrong number of arguments passed

### ISSUE THREAD

**Issue by stefanradziuk** [WebAssembly/wabt#2242]
wasm-interp crashes with no error message when the wrong number of arguments is given.

For example, when running the identity function:

```wasm
(module
  (func (export "id") (param i32) (result i32) (local.get 0))
)
```

```
% wasm-interp identity.wasm --run-export=id
[1]    102759 segmentation fault (core dumped)  wasm-interp identity.wasm --run-export=id

% wasm-interp identity.wasm --run-export=id --argument=i32:7
id(i32:7) => i32:7

% wasm-interp identity.wasm --run-export=id --argument=i32:7 --argument=i32:8
[1]    102811 abort (core dumped)  wasm-interp identity.wasm --run-export=id --argument=i32:7 --argument=i32:8
```

Version:
```
% wasm-interp --version
1.0.33
```

**Comment by stefanradziuk** [comment]
> Do you also get that error message? If yes, you meant to have a more clear error message?

No, I get an abort / segfault with no error message.

Interesting re 1.0.32. I don't have access to a 1.0.32 build to test it right now but I'll try to later.

I'm using the [1.0.33 build from nixpgs](https://github.com/NixOS/nixpkgs/blob/58ca986543b591a8269cbce3328293ca8d64480f/pkgs/development/tools/wabt/default.nix#L16) if that makes a difference. It doesn't look to me like they're disabling assertions at compile time or anything like that.

**Comment by raphamorim** [comment]
Yes, i believe you are right. as far I understand assertion isn't available in "release" builds. Hmm, I believe makes more sense launch an error along with the assert.

**Comment by raphamorim** [comment]
@stefanradziuk added a fix for that in the #2250 PR

**Comment by stefanradziuk** [comment]
Thanks!
